Source/loadsave.cpp

```diff
@@ -865,9 +865,20 @@ void LoadItem(LoadHelper &file, Item &item)
 	GetItemFrm(item);
 }
 
-void LoadPremium(LoadHelper &file, int i)
+void LoadPremiumItems(LoadHelper &file)
 {
-	LoadAndValidateItemData(file, PremiumItems[i]);
+	// Resize the vector to the expected number of items in the save file
+	Blacksmith.items.resize(giNumberOfSmithPremiumItems);
+
+	for (int i = 0; i < giNumberOfSmithPremiumItems; ++i) {

```

Comment:
Are we sure that `giNumberOfSmithPremiumItems` is always the same number as `Blacksmith.itemCount`?